### DISCUSSION TITLE
How to do the correct request? Error: SoapClient::$request must not be accessed before initialization

### DISCUSSION THREAD

**Discussion by supermavster** [CodeDredd/laravel-soap#211] [Q&A] [answered]
Hi I have a problem call a method, but I don't have idea how to do it that:

The error is:

```
Typed property CodeDredd\\Soap\\SoapClient::$request must not be accessed before initialization
```

I try to create this petition **WITHOUT SSL** Certification: https://xxxxxxxx/xxx/xxx/xxx?wsdl

```xml
<soapenv:Envelope
	xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
	xmlns:ser="http://service.cows.gnconsult.com/">
	<soapenv:Header/>
	<soapenv:Body>
		<ser:cons_saldo_cliente>
			<cliente>1234</cliente>
			<entidad>12345</entidad>
		</ser:cons_saldo_cliente>
	</soapenv:Body>
</soapenv:Envelope>
```

I create based in the other comments this code:

```php
$urlWSDL = 'https://xxxxxxxx/xxx/xxx/xxx?wsdl';

$client = Soap::withGuzzleClientOptions([
            'allow_redirects' => RedirectMiddleware::$defaultSettings,
            'http_errors' => true,
            'decode_content' => true,
            'verify' => false,
            'cookies' => false,
            'idn_conversion' => false,
        ])
            ->baseWsdl($urlWSDL)
            ->withOptions([
                'ser' => 'http://service.cows.gnconsult.com/',
            ])
            ;

        $response = $client->call('ser:cons_saldo_cliente', [
                'cliente' => '1234',
                'entidad' => '12345',
            ])
            ->throw()
            ->json();

        return $response;
```

But in diferente case is the same answere:

```Typed property CodeDredd\\Soap\\SoapClient::$request must not be accessed before initialization```

Please help me!

**Comment by CodeDredd** [comment, marked as answer]
@supermavster Heyho

do you have an stacktrace for me? I think i know the reason. I am fixing it right away but for fixing it the right way i need to know more.

**Reply by supermavster** [reply]
The problem is that this phrase:

```
...
xmlns:ser="http://service.cows.gnconsult.com/">
```

No exist this group and the library not have the function to create or add or append and extra `xmlns` (I think...).

Because when I try to call: 
```
<ser:cons_saldo_cliente>
....
```

not exist this function "ser"

**Reply by CodeDredd** [reply]
You need to create a custom middleware. Take a look at ``CisDhlMiddleware.php``. 
You have alle the option creating custom xml setup with ``https://github.com/veewee/xml`` there.
